lib/fetch/index.js

```diff
@@ -1990,4 +1991,72 @@ function httpNetworkFetch (
   })
 }
 
+async function fileFetch (fetchParams) {
+  const { request } = fetchParams
+  const url = requestCurrentURL(request)
+  const context = this
+  if (request.method !== 'GET') {
+    return makeNetworkError(`Fetching files only supports the GET method. Received ${request.method}`)
+  }
+
+  let stream
+  let file
+
+  try {
+    file = await fs.promises.open(url)
+    if (context.terminated) {
+      if (context.terminated.aborted) {
+        throw new AbortError()
+      } else {
+        throw new Error('terminated')
+      }
+    }
+    stream = file.createReadStream({ autoClose: true })
+  } catch (originalError) {
+    if (stream) {
+      stream.destroy(originalError)
+    } else if (file) {
+      try {
+        await file.close()
+      } catch (closeError) {
+        return createAggregateNetworkError(originalError, closeError)
+      }
+    }
+    return makeNetworkError(originalError)
+  }
+
+  try {
+    function onTerminated () {
+      const aborted = context.terminated.aborted
+      if (aborted) {
+        response.aborted = true
+        stream.destroy(new AbortError())
+      } else {
+        stream.destroy(new TypeError('terminated'))
```

Comment:
Not specific to these PR but these errors should have codes?

(That seems orthogonal though)

Reply:
I looked at what's done in other methods.